Charter.md

```diff
@@ -1,70 +1,69 @@
-patent section applies to, it is helpful to describe the scope in a way
-that it is clear what types of technologies will be defined in
-specifications, as opposed to adoption by reference or underlying
-technology not defined in the proposed spec. Key use cases are often
-helpful in describing scope. If no specifications will be defined in the
-group that the CLA patent section applies to, the charter should clearly
-state that. A clear scope is particularly important where patent
-licensing obligations may apply.}*
+The common API developed by the Community Group will be defined as a set of traits and/or lightweight type,
+that other crates are encouraged to implement and reuse,
+so that data values produced by an implementation can be reused by another one.
+
+For example, the triples produced by a [Turtle]
+parser, provided that they implement a trait defined by this Community Group,
+could be ingested by any independently developed implementation of a graph store.
+Similarly, if the latter implements the appropriate traits,
+it could be checked by any independently developed [SHACL] validator.
+
+As much as possible,
+the crates developed by the Community Group should aim at fulfilling the
+*[zero-cost abstraction]* motto:
+the provided traits should ideally be generic enough,
+to be directly implementable by any specific implementation,
+regardless of its internal design
+(as opposed to requiring a [wrapper type]).
+However, overly generic code can be hard too use in practice,
+so the Community Group will have to strike a balance between genericity and usability.
+
+The Community Group may also develop utility [library crates], i.e.,
+crates providing common types or functions expected to be pervasively used in the implementations of the common API.
+For example, the Community Group may produce a crate for parsing and resolving IRIs.
 
 ### Out of Scope
 
-*{TBD: Identify topics known in advance to be out of scope}*
+* develop an complete implementation of RDF-related standards
 
 ## Deliverables
 
+### Rust Crates
+
+The main deliverables of the community group are Rust [library crates],
+which should eventually be published on the standard repository https://crates.io/.
+
 ### Specifications
 
-*{TBD: Provide a brief description of each specification the group plans
-to produce. Where an estimate is possible, it can be useful to provide
-an estimated schedule for key deliverables. As described below, the
-group may later modify the charter deliverables. if no specifications,
-include: \"No Specifications will be produced under the current
-charter.\"}*
+No Specifications will be produced under the current charter.
 
 ### Non-Normative Reports
 
-The group may produce other Community Group Reports within the scope of
-this charter but that are not Specifications, for instance use cases,
-requirements, or white papers.
+The community group MAY produce a user manual, or other kind of documentation,
```

Comment:
This is the only part I disagree with.
In my experience, a complete documentation (detailed comments on all public methods and runnable examples) on docs.rs is essential for Rust libraries. It doesn't need to be an external user manual but I would prefer if rustdoc comments are required as a "MUST" and not "MAY".

Reply:
I agree that code documentation is a MUST. However in my understanding this sentence is about other documentation or manual, the docs.rs documentation being part of the code itself and published alongside it.

What about adding a sentence like " public APIs MUST be extensively documented using doc comments and this documentation published on docs.rs like any other Rust crate library"?

Reply:
This is a great suggestion, I just would add that there should be clear  examples that compile as given to accomplish basic tasks, or is that not applicable for the intended libraries as the libraries will only be APIs for others in the end and not perform specific tasks?

What I mean is that sometimes all the functions in a library are documented but you still don't know what you should do to create an X or load a Y or validate a Z.

Reply:
thanks @KonradHoeffner for this feedback.

@Tpt is right: this item is about *additional* documents. It goes without saying, in my opinion, that the code must be properly documented on docs.rs . But it probably does not hurt to say it.

I'll make a change to that effect.

NB: also, it was a mistake from me to use BCP14-style "MAY" in capital letters -- charters usually don't use this convention, so I'll fix that too.

Reply:
Done in e0ade01  . Note that I didn't mention docs.rs, because
* AFAIK, it is done automatically when releasing on crates.io
* should another service emerge and replace docs.rs, we don't need to change the charter :-)
* we may also want to publish a live documentation using GH pages, but we can discuss this later.